service/worker/archiver/archiver.go

```diff
@@ -115,57 +114,15 @@ func handleRequest(ctx workflow.Context, logger log.Logger, metricsClient metric
 	}
 	actCtx := workflow.WithActivityOptions(ctx, ao)
 	uploadSW := metricsClient.StartTimer(metrics.ArchiverScope, metrics.ArchiverUploadWithRetriesLatency)
-	var result *uploadResult
-	err := workflow.ExecuteActivity(actCtx, uploadHistoryActivityFnName, request).Get(actCtx, &result)
-	if err == nil && result != nil {
-		err = errors.New(result.ErrorWithDetails)
-	}
+	err := workflow.ExecuteActivity(actCtx, uploadHistoryActivityFnName, request).Get(actCtx, nil)
 	if err != nil {
-		logger.Error("failed to upload history, will delete all uploaded blobs and moving on to deleting history without archiving", tag.Error(err))
+		logger.Error("failed to upload history, will move on to deleting history without archiving", tag.Error(err))
```

Comment:
"failed to archive history" its possible archiving history does not mean uploading.